doc/module-system/module-system.chapter.md

```diff
@@ -1,13 +1,211 @@
 # Module System {#module-system}
 
+::: {.note}
+This chapter is new and not complete yet. For a gentle introduction to the module system, in the context of NixOS, see [Writing NixOS Modules](https://nixos.org/manual/nixos/unstable/index.html#sec-writing-modules) in the NixOS manual.
+:::
+
 ## Introduction {#module-system-introduction}
 
+::: {.note}
+This chapter is new and not complete yet.
+:::
+
 The module system is a language for handling configuration, implemented as a Nix library.
 
-Compared to plain Nix, it adds documentation, type checking and composition or extensibility.
+It was historically known as the NixOS module system.
 
-::: {.note}
-This chapter is new and not complete yet.
+Its purpose is to "extend" the Nix language with
+
+- **dynamic type checking** to find mistakes early on, improving error messages;
```

Comment:
```suggestion
- **dynamic type checking** to catch and locate mistakes early on;
```

It would be a bit tone deaf to speak about improving error messages from the position of official documentation.